skbeam/core/accumulators/arrayProjector.py

```diff
@@ -0,0 +1,155 @@
+
+        return
+
+    def __call__(self, image):
+        """
+        Bin pixel intensities.
+        
+        Parameters
+        ----------            
+        image : np.ndarray
+            The intensity at each pixel
+        Returns
+        -------
+        bin_values : np.ndarray
+            The average intensity in the bin
+        """
+
+        if not (image.shape == self.bin_values.shape):
+            raise ValueError('`image` and `bin_values` must have the same shape',image.shape,self.bin_values.shape)
+
+        if self.weights is None:
+            weights = image.flatten()
+        else:
+            weights = image.flatten() * self.weights.flatten()
+            if not (image.shape == self.weights.shape):
+                raise ValueError('`image` and `weights` must have the same shape')
+
+        bin_values = np.bincount(self._bin_assignments.flatten(), weights=weights)
+
+        if self.norm:
+            bin_values /= self._normalization_array
+
+        assert bin_values.shape[0] == self.nbins
+
+        return bin_values
+
+    @property
+    def bin_centers(self):
+        """
+        Returns:
+        --------
+        bin_centers : ndarray, float
+            The center of each bin.
+        """
+        return (np.arange(self.nbins) + 0.5) * self.bin_width + self.bin_values.min()
+
+class RadialProjector(ArrayProjector) :
+    """ 
+    Project a 2D image onto a radial axis
+    """
+    
+    def __init__(self, rows, columns, xc=None, yc=None, rmin=None, rmax=None, phimin=None, phimax=None, nbins=101, weights=None, norm=True):
+        """
+        Parameters:
+        -----------
+        rows,columns:  shape of image to be projected
+        xc,yc:         location (in pixels) of origin (default: center of image)
+        rmin,rmax:     radial range to include in projection, in pixels (default: no limits)
+        phimin,phimax: phi range to include in projection, in degrees (default: no limits)
+        nbins:         number of bins in projected histogram
+        weights:       np.ndarray.  weight to be applied to each pixel in image.  this can
+                       be used as a mask if the weight is set to zero.
+        norm:          boolean indicating whether bin entries in the projected histogram should be divided
+                       by weights (number of pixels, in the case where the weights are 1).
+        """   
+        # flip these to make it more intuitive for users who will use
+        # this with arrays (matrices).  numpy.meshgrid (version 1.6) uses
+        # cartesian (non-matrix) coordinates, so we need to do the flip
+
+        xsize   = columns
+        ysize   = rows
+
+        xc    = xsize/2                  if xc    is None else xc
+        yc    = ysize/2                  if yc    is None else yc
+        rmin  = 0                        if rmin  is None else rmin
+        rmax  = math.sqrt(xc**2 + yc**2) if rmax  is None else rmax
```

Comment:
You're right.  I thought about this logic a little more, and I think we don't need these rmax/rmin lines anymore.  Will remove them.